### DISCUSSION TITLE
C API feature request: Add support for BLOB columns in table functions

### DISCUSSION THREAD

**Discussion by ak-coram** [duckdb/duckdb#7977] [Ideas]
I'm working to extend table function support in my DuckDB wrapper with more column types: https://github.com/ak-coram/cl-duckdb/issues/43

I imagine support for blob values could be very similar to how strings currently work where we have `duckdb_vector_assign_string_element` and `duckdb_vector_assign_string_element_len`: couldn't the same thing be done for BLOB columns?

If I'm missing something please let me know!

**Comment by solidsnack** [comment]
I'm working in Rust and this would help me, too!

**Comment by Mytherin** [comment]
This has been implemented in https://github.com/duckdb/duckdb/pull/10748, you can now use `duckdb_vector_assign_string_element_len` for blobs directly as well
